src/Workspaces/Core/Portable/Rename/IConflictResolution.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Rename
+{
+    /// <summary>
+    /// Represent the result of rename engine. It could either be a succesful <see cref="ConflictResolution"/> or failed <see cref="FailedConflictResolution"/>
```

Comment:
i'm really not a fan of using inheritance in this fashion.  we could just have a normal struct/class, and a boolean on it specifying what sort it is.  the boolean would indicate which parts of the api are null or not.

Reply:
Do you prefer something like
```
partial class ConflictResolution
{
      [MemberNotNullWhen(true, nameof(FieldA), nameOf(FieldB))]
      bool IsSuccessful { get; }
      // Some other fields
}
```

Reply:
: ) I thought the Inheritance pattern is our standard way to simulate discrimination union in F#. But I am fine with both ways

Reply:
I change the implementation to the way not using inheritance